crml/cennzx-spot/src/types.rs

```diff
@@ -16,87 +16,136 @@
 //!
 //! CENNZX-SPOT Types
 //!
-use core::convert::TryInto;
-use uint::construct_uint;
-construct_uint! {
-	/// 128-bit unsigned integer.
-	pub struct U128(2);
+
+use codec::{Decode, Encode};
+use core::{
+	convert::{From, Into, TryFrom},
+	marker::PhantomData,
+};
+use support::decl_error;
+
+pub use primitive_types::U256 as HighPrecisionUnsigned;
+pub use u128 as LowPrecisionUnsigned;
+
+/// A trait for values which hold an implicit scale factor that needs to be taken into account in calculations
+pub trait Scaled {
+	const SCALE: LowPrecisionUnsigned;
 }
 
-construct_uint! {
-	/// 256-bit unsigned integer.
-	pub struct U256(4);
+/// Per millionth of unit price
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum PerMill {}
+impl Scaled for PerMill {
+	const SCALE: LowPrecisionUnsigned = 1_000_000;
 }
 
-use codec::{Compact, CompactAs, Decode, Encode};
+/// Per thousandth of unit price
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum PerMilli {}
+impl Scaled for PerMilli {
+	const SCALE: LowPrecisionUnsigned = 1_000;
+}
```

Comment:
I know I'm not strictly a reviewer on this, but I would suggest making `PerMill` and `PerMilli` more distinct from eachother.
It isn't obvious that Mill is abbreviating a Million. Also the two words are effectively "PerM" plus a bunch of lines - I remember reading a good writeup in Code Complete in what naming anti-patterns to avoid.

Anyway, I would suggest remating these to `PerMilli` and `PerMillion` to avoid accidental mixing of the two values =)

Reply:
I don't mean to push the point too much, but a quick look online for the meaning of the word Mill gives:

![Uploading image.png…]()

Reply:
Sounds a good suggestion to me. However, because we have had the notions of Permill, Percent and even Perbill not limited to this place in the code, I would like to know if @holygits would prefer this change or not.